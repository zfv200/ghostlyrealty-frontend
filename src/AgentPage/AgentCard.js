import React from 'react'

const AgentCard = (props) => {

  return (
    <div className="agent-card">
      <div class>
        <h1>{props.name}</h1>
        <img src={props.image}/>
      </div>
      <h3>{props.title}</h3>
    </div>
  )
}

export default AgentCard
