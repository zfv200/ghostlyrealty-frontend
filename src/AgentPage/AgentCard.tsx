import React from 'react'


interface AgentStatelessProps {
  username: string,
  image: string,
  motto: string
}

const AgentCard: React.SFC<AgentStatelessProps> = ({ username, image, motto }) => {
  return (
    <div className="agent-card ph5 flex bb mb3 pb3" style={{width: "100%"}}>
      <img className="fl p2" alt={`${username}'s headshot'`} src={image}/>
      <div className="fr w-100 ml5">
        <h1 className="f3 mb5" >{username}</h1>
        <h3 className="i">{motto}</h3>
      </div>
    </div>
  )
}

export default AgentCard
