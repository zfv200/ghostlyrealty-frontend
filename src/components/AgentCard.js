import React from 'react'

const AgentCard = (props) => {

  return (
    <div>
      <h1>{props.name}</h1>
      <img src={props.image}/>
      <h3>{props.title}</h3>
    </div>
  )
}

export default AgentCard
