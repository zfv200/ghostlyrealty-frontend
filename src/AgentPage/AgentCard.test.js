import React from 'react'
import ReactDOM from 'react-dom'
import AgentCard from './AgentCard.tsx'

import { shallow } from 'enzyme'

const agent = {
  username: 'King Hamlet',
  age: 500,
  image: 'https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg',
  motto: "Lend thy serious hearing--for your next move!"
}

it('renders without crashing', ()=>{
  const component = shallow(<AgentCard {...agent}/>)

  expect(component).toMatchSnapshot()
})
