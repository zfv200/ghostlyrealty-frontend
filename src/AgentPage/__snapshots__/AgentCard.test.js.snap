// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AgentCard
    age={500}
    image="https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg"
    motto="Lend thy serious hearing--for your next move!"
    username="King Hamlet"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1>
            King Hamlet
          </h1>
          <img
            alt="King Hamlet's headshot'"
            src="https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg"
          />
        </div>,
        <h3>
          Lend thy serious hearing--for your next move!
        </h3>,
      ],
      "className": "agent-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              King Hamlet
            </h1>,
            <img
              alt="King Hamlet's headshot'"
              src="https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "King Hamlet",
            },
            "ref": null,
            "rendered": "King Hamlet",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "King Hamlet's headshot'",
              "src": "https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lend thy serious hearing--for your next move!",
        },
        "ref": null,
        "rendered": "Lend thy serious hearing--for your next move!",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1>
              King Hamlet
            </h1>
            <img
              alt="King Hamlet's headshot'"
              src="https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg"
            />
          </div>,
          <h3>
            Lend thy serious hearing--for your next move!
          </h3>,
        ],
        "className": "agent-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                King Hamlet
              </h1>,
              <img
                alt="King Hamlet's headshot'"
                src="https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "King Hamlet",
              },
              "ref": null,
              "rendered": "King Hamlet",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "King Hamlet's headshot'",
                "src": "https://i.ytimg.com/vi/IBQucj2hea4/hqdefault.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lend thy serious hearing--for your next move!",
          },
          "ref": null,
          "rendered": "Lend thy serious hearing--for your next move!",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
